verilog/tools/syntax/verilog_syntax_test.sh

```diff
@@ -94,6 +122,35 @@ EOF
 diff --strip-trailing-cr -u "$MY_EXPECT_FILE" "$MY_OUTPUT_FILE".filtered || \
   { echo "stderr differs." ; exit 1 ;}
 
+################################################################################
+echo "=== Test reading stdin, print errors. --export_json"
+
+"$syntax_checker" --export_json - > "$MY_OUTPUT_FILE" 2>&1 <<EOF
+module 1;
+endmodule
+module 2;
+endmodule
+EOF
+
+status="$?"
+[[ $status == 1 ]] || {
+  "Expected exit code 1, but got $status"
+  exit 1
+}
+
+json_canonicalize > "$MY_EXPECT_FILE" <<EOF
+{"-":{
+  "errors": [
+      {"line": 0, "column": 7, "phase": "parse", "text": "1" },
+      {"line": 1, "column": 0, "phase": "parse", "text": "endmodule" },
+      {"line": 3, "column": 0, "phase": "parse", "text": "endmodule" }
+  ]}}
+EOF
+
+tr '\r\n' '\n' < $MY_OUTPUT_FILE | json_canonicalize > "${MY_OUTPUT_FILE}.1"
```

Comment:
FYI, there's a too called dos2unix that helps with newline conversion as well.

Reply:
I wanted to keep the number of different tools used as little as possible, as this would run on various operating systems.